ministryofjustice/correspondence_tool_staff
# CT-3039 showing deleted case in case transitions

## Description

This PR is to make the case-history displayable after a historic linked case is deleted, so changes I made is to make the whole story around the deletion side completed
- Check whether there are some linked cases before deletions
- Show different wording based on whether the historic linked case has been deleted or not
 
## Self-review checklist


### Screenshots

Before 
Grey death screen

After  
1 - Extra checking when a case is deleted 
<img width="1007" alt="error_message_due_to_linked_to_a_case" src="https://user-images.githubusercontent.com/26111671/95188977-270c4780-07c5-11eb-9f2b-290688b23592.png">

2 - For the case which has a linked case, the case history will look like below if the linked case hasn't been deleted
<img width="851" alt="updated_case_history_in_the_linked_case" src="https://user-images.githubusercontent.com/26111671/95189611-00024580-07c6-11eb-9f5d-2d3a957b71df.png">

3- After the linked case is deleted, the case history will look like
<img width="1034" alt="updated_case_history_after_deletion" src="https://user-images.githubusercontent.com/26111671/95189634-05f82680-07c6-11eb-8c72-f3f1de22c73c.png">

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3039
### Deployment


### Manual testing instructions

1 - Login as London user
2 - Pick up an example FOI/SAR case 
3 - Pick up an case to link to the case in step 2 
4 - Load the case detail for the case in step 2
5 - Click "Link a case" in linked cases section and type the number of case from step 3, then click 'Add link'
6 - Check case details for the case in step 3 and will see there is new entry in case history from step 5
7 - Come back to the cases in step 2 and choose "Delete case"
8 - Click "Delete", an error message will pop up saying the related case links are not empty, you need to remove link first 
9 - Following the instruction, click 'Cancel' to come back to the case detail
10- Remove the linked case by clicking 'Remove link' from the linked cases section
11 - Check the case history again for case in step 3, and a new entry should be added from step 10 and it shows the number of case in step 2 
12 - Delete the case from step2 by following step 8 
13 - Check the case history again for case in step 3, and the wording for the entry from step 11 should be changed to show the id of case only plus some extra note
